### ISSUE TITLE
Why are the number of labels and the number of image feature regions unequal in the CaptionTensorizer Coco-caption

### ISSUE THREAD

**Issue by ZuoJiaxing** [microsoft/Oscar#15]
https://github.com/microsoft/Oscar/blob/a9013bb7dda35a63856d1cebd16eeeeb73615e5c/oscar/run_captioning.py#L195
Hello, could you please explain why the number of labels(text_b) is not equal to the number of image feature regions?  It's a little bit weird from my point of view.

**Comment by xiyinmsu** [comment]
there is no need to have 1 to 1 correspondence between image region features and object labels. We tried labels from different models and features from a single object detector model.

**Comment by ZuoJiaxing** [comment]
So this is different from the pretraining where you need the 1 to 1 correspondence to formulate the conservative loss (whether the labels are polluted?)

**Comment by xiyinmsu** [comment]
The contrastive loss is applied on the image level. If we pollute the tags, we will randomly choice all tags from a different image to replace the right tags. So there is no need for 1 to 1 correspondence.

**Comment by ZuoJiaxing** [comment]
Thanks! Why not a masked object label prediction loss? What's your intuition?

**Comment by xiyinmsu** [comment]
we have a masked label prediction loss during pre-training.

**Comment by ZuoJiaxing** [comment]
But it seems that you have not mentioned this in your paper.

**Comment by xjli** [comment]
See page 5, 

"We define the discrete token sequence as h = [w, q], and apply the Masked Token Loss (MTL) for pre-training".
